### ISSUE TITLE
Code formatting

### ISSUE THREAD

**Issue by schriftgestalt** [microsoft/VisualTrueType#11]
The code format is a bit messy. I could make a clang-format file that is close to what most of the code looks like?

**Comment by paullinnerud** [comment]
The code is very messy. Some of it started on Mac decades ago, went to Windows and now back to Mac again. Inconsistent usage of tabs and spaces, etc. A clang-format file would be nice. I believe VSCode supports those or supports an extension that supports. Probably does not help those who use Visual Studio though.

**Comment by schriftgestalt** [comment]
I think my format file is already in HinTaks repo. I didn’t apply it, yet. So you can try and maybe edit it before we apply it to the whole source.

**Comment by HinTak** [comment]
Yes, the clang-format file is already in mine. I would prefer that the more obvious warnings get fixed first before large cosmetic changes though. I know I broke my own preference, changing all the mac-roman encoded comments and win 1252 comments into unicode (this is a "cosmetic" change in my book) but it is an extension of the non-ascii wide character literal fix, which is a code fix... 

It was quite surprising to find non-ascii comments of the mac-roman encoded type and win 1252 type, (and in one file, both types in different sections). It certainly suggests a very long and pre-unicode history of the code base, as well as it having passed between mac and windows.

**Comment by HinTak** [comment]
The code currently generates a few "misleading indentation" / "dangling-else" warnings - I looked at one of them (the one in getopt.h) , and it is not entirely clear that it is mis-intented, instead of braces being mistakenly removed (when bits were being removed from full vtt to vttcompile). So that's another reason not to reindent, until those are fixed. Re-indent at the stage, if the code is actually broken because braces were mis-removed, would make it more difficult to fix.

Basically I am very against large cosmetic changes before almost all the warnings are addressed/looked at.
